ARCHITECTURE.md

```diff
@@ -0,0 +1,58 @@
+# Architecture
+
+This document describes high-level architecture of signs-ui. The goal of this document is to
+provide a useful orientation to a new contributor to the codebase.
+
+## Application description
+
+Signs UI is a web app used in primarily two ways:
+
+1. To monitor the current state of countdown signs throughout the system. (The "viewer")
+2. To control the signs in certain ways. (The "configuration")
+
+Like Realtime Signs, throughout this app when we talk about a "sign" it could actually represent
+multiple physical signs in the field, all behaving identically. In ARINC terminology a sign is a
+"zone" at a station and is the minimally granular unit that we have control over.
+
+## Architecture of the "viewer"
+
+The applications that drive the signs throughout the system, Realtime Signs for subway and
+TransitwayEngine for bus, do so by sending HTTP POST requests to the ARINC head-end server. The
+two apps, upon a successful post to ARINC, also send the exact same request to signs-ui, which is
+handled by the `SignsUiWeb.MessagesController`.
+
+Signs-UI tries to simulate the behavior of the ARINC head end server. The
+`SignsUi.Messages.SignContent` module is a parser that handles the particular format of
+`MsgType=SignContent` messages that ARINC can receive. It does not handle `AdHoc` or `Canned`
+messages, which are generally used for audio.
+
+The received messages get stored in memory in the `SignsUi.Signs.State` GenServer. This means when
+signs-ui starts up, the known contents of every sign is blank. However, since message expirations
+are generally set at 2-3 minutes, our apps will send a POST to every sign in the system relatively
```

Comment:
Ah, it refers to Realtime Signs and Transitway Engine. Since over the course of 2-3 minutes, those apps will send at least one message to every sign they know about, it only takes 2-3 minutes for signs-ui to build up its state again. I'll update the text to say Realtime Signs and Transitway Engine instead of "apps".